src/html.js

```diff
@@ -125,4 +125,92 @@ const htmlGenerator = function* ({ raw: literals }, ...expressions) {
   }
 };
 
-export { html, htmlGenerator };
+/**
+ * @param {{ raw: string[] }} literals Tagged template literals.
+ * @param {...any} expressions Expressions to interpolate.
+ * @yields {string} The HTML strings.
+ */
+const htmlAsyncGenerator = async function* ({ raw: literals }, ...expressions) {
+  let index = 0;
+
+  for (; index !== expressions.length; ++index) {
+    let literal = literals[index];
+    let expression;
+
+    if (expressions[index] instanceof Promise) {
+      expressions[index] = await expressions[index];
+    }
+
+    if (typeof expressions[index] === "string") {
+      expression = expressions[index];
+    } else if (
+      expressions[index] === undefined ||
+      expressions[index] === null
+    ) {
+      expression = "";
+    } else {
+      if (
+        typeof expressions[index][Symbol.iterator] === "function" ||
+        typeof expressions[index][Symbol.asyncIterator] === "function"
+      ) {
+        const isRaw =
+          literal.length !== 0 && literal.charCodeAt(literal.length - 1) === 33;
+
+        if (isRaw) {
+          literal = literal.slice(0, -1);
+        }
+
+        if (literal.length) {
+          yield literal;
+        }
+
+        for await (const value of expressions[index]) {
+          if (typeof value === "string") {
+            expression = value;
+          } else if (value === undefined || value === null) {
+            continue;
+          } else if (
+            typeof value[Symbol.iterator] === "function" ||
+            typeof value[Symbol.asyncIterator] === "function"
+          ) {
+            expression = "";
+
+            for await (const innerValue of value) {
+              expression += innerValue ?? "";
+            }
+          } else {
+            expression = `${value}`;
+          }
+
+          if (expression.length) {
+            if (!isRaw) {
+              expression = expression.replace(escapeRegExp, escapeFunction);
+            }
+
+            yield expression;
```

Comment:
This is problematic. You should be yielding things as they come inside the for..await loop (and escaping there, too)

Reply:
I fixed it but the code is not looking good in my opinion, will refactor it a little more